File: coin-changer/coin-changer.rb

# Person has a bunch of pennies
# Person wants to convert them into the least amount of quarters, nickels, and dimes.
# Any remaining pennies are counted too
#_______________________________
# Get total amount of pennies
# Reduce by quarters
  # Push quarters into an array
# Reduce by dimes
  # Push dimes into an array
# Reduce by nickels
  # Push nickels into an array
# Push remaining pennies into array
# Count number of each coin in array

class CoinChanger
  def calculate(amount)
    coins = []
    while amount >= 25
      coins << "25"
      amount -= 25
    end

    while amount >= 10
      coins << "10"
      amount -= 10
    end

    while amount >= 5
      coins << "5"
      amount -= 5
    end

    while amount >= 1
      coins << "1"
      amount -= 1
    end

    coins.join(", ")
  end
end

File: coin-changer/coin-changer.js

function coinChanger(amount) {
  var original = amount
  var coins = []
  var types = [25, 10, 5, 1]
  for (var i = 0; i < types.length; i++) {
    while (amount >= types[i]) {
      coins.push("" + types[i]);
      amount -= types[i];
    };
  };
  console.log(original + ":\t" + coins.join(", "));
}

coinChanger(25);
coinChanger(10);
coinChanger(5);
coinChanger(1);
coinChanger(39);
coinChanger(99);
coinChanger(64);
